app/services/reports_tar_reader.rb

```diff
@@ -10,15 +10,22 @@ def initialize(file)
   end
 
   def reports
-    tar_extract = Gem::Package::TarReader.new(Zlib::GzipReader.new(@file))
-    tar_extract.rewind # The extract has to be rewinded after every iteration
-    tar_extract.map do |file|
-      file.read if file.header.name.match(/oscap_results/)
+    extract.map do |file|
+      filename = File.basename(file.header.name)
```

Comment:
this one seems to be truncated :frowning_face:

Reply:
```
struct tarfile_entry_posix {
  char name[100];     # ASCII + (Z unless filled)
            ^^^
```

Reply:
Okay, so I've dropped the filename gathering and rather changed it to take profile id from parsed content.